### PR TITLE
Issue 151220/add current timezone offset variable

### PR DESCRIPTION
**Pull request by MonadChains**
Solves #151220

### CODE REVIEW

File: src/vs/editor/contrib/snippet/browser/snippetVariables.ts [hunk 1]
```diff
@@ -292,6 +293,14 @@ export class TimeBasedVariableResolver implements VariableResolver {
 			return TimeBasedVariableResolver.monthNamesShort[this._date.getMonth()];
 		} else if (name === 'CURRENT_SECONDS_UNIX') {
 			return String(Math.floor(this._date.getTime() / 1000));
+		} else if (name === 'CURRENT_TIMEZONE_OFFSET') {
+			const rawTimeOffset = this._date.getTimezoneOffset();
+			const sign = rawTimeOffset < 0 ? '-' : '+';
```
Comment (mattjohnsonpint) [inline src/vs/editor/contrib/snippet/browser/snippetVariables.ts:298]:
```suggestion
			const sign = rawTimeOffset > 0 ? '-' : '+';
```

Sign should be inverted.  ISO 8601 is positive *east* of GMT while `getTimezoneOffset` is positive *west* of GMT.  Zero should always be positive.

Comment (mattjohnsonpint) [reply src/vs/editor/contrib/snippet/browser/snippetVariables.ts:298]: Examples: `Asia/Kolkata` should be `+05:30`.  `America/Phoenix` should be `-07:00`.

**Comment by mattjohnsonpint** [comment]
Please also add tests, or update these ones.  Thanks.
https://github.com/microsoft/vscode/blob/4acf2d9b46b75748ae687cf3b2952a0799679873/src/vs/editor/contrib/snippet/test/browser/snippetVariables.test.ts#L293-L339

File: src/vs/editor/contrib/snippet/browser/snippetVariables.ts (hunk 1 above)
Comment (MonadChains) [reply src/vs/editor/contrib/snippet/browser/snippetVariables.ts]: Thanks for spotting the error, I've fixed it.

**Comment by MonadChains** [comment]
@mattjohnsonpint I've add the new variable to the tests

**Review by mattjohnsonpint — APPROVED** [review]

**Comment by mattjohnsonpint** [comment]
LGTM, but I'm not a maintainer of this repo.

**Comment by redactedscribe** [comment]
@jrieken Any chance we could get this PR approved? Would love to see this in VS Code.

**Review by jrieken — APPROVED** [review]

**Comment by jrieken** [comment]
Yeah, this slipped my inbox.

**Review by ulugbekna — APPROVED** [review]

**Pull request merged**